src/utils/logging/serializeResponse.ts

```diff
@@ -1,11 +1,16 @@
 import { flattenHeadersObject, headersToObject } from 'headers-polyfill'
 import type { SerializedResponse } from '../../setupWorker/glossary'
 
-export function serializeResponse(source: Response): SerializedResponse<any> {
+export async function serializeResponse(
+  response: Response,
+): Promise<SerializedResponse<string>> {
   return {
-    status: source.status,
-    statusText: source.statusText,
-    headers: flattenHeadersObject(headersToObject(source.headers)),
-    body: source.body,
+    status: response.status,
+    statusText: response.statusText,
+    headers: flattenHeadersObject(headersToObject(response.headers)),
+    // Serialize the response body to a string
+    // so it's easier to process further down the chain in "prepareResponse" (browser-only)
+    // and "parseBody" (ambiguous).
+    body: await response.text(),
```

Comment:
## Beware: Not an actual `Response`!

Using a normal `Response` here, we'd have to clone it first so every consumer down the chain would still be able to read it. But this is _not_ a Fetch API `Response` instance here. It's our internal, custom thing since as per 1.2 we are still running in ambiguous, old environments without the global Fetch API. 

Won't be a thing since 2.0.

Reply:
Consequentially, this doesn't block the consumers from reading the response body again.

Reply:
@kettanaito after updating to 1.2.2 it looks like i'm getting errors here 

```
Uncaught (in promise) TypeError: Failed to execute 'text' on 'Response': body stream already read
    at serializeResponse (index.js:1616:1)
    at Emitter.<anonymous> (index.js:1660:1)
    at Emitter.onceListener (Emitter.js:40:1)
    at Emitter.js:75:1
    at Array.forEach (<anonymous>)
    at Emitter.emit (Emitter.js:74:1)
    at source.emit (index.js:2036:1)
    at index.js:1756:1
    at ServiceWorkerContainer.<anonymous> (index.js:2140:1)
serializeResponse @ index.js:1616
(anonymous) @ index.js:1660
onceListener @ Emitter.js:40
(anonymous) @ Emitter.js:75
emit @ Emitter.js:74
source.emit @ index.js:2036
(anonymous) @ index.js:1756
(anonymous) @ index.js:2140
await in (anonymous) (async)
onceListener @ Emitter.js:40
(anonymous) @ Emitter.js:75
emit @ Emitter.js:74
source.emit @ index.js:2036
(anonymous) @ index.js:1756
(anonymous) @ index.js:2140

```

Reply:
It seems that `quiet: true` gets rid of those issues though, since it no longer tries to read that response twice?

Reply:
@mattcosta7, do you have a reproduction case for this? 

I was concerned about this too but the `response` we use as of 1.x is _not_ a Fetch AIP Response. It doesn't even have the `.clone()` method and is meant purely for the internal representation of responses. Thus, there's no issue of reading the body multiple times. 

I'd like to look closer at this.

Reply:
I'll try to get something together when I can.  it seems to happen f

```
rest.post('<url>', async (req, res, ctx) => {
   return res(ctx.status(200))
})
```

that alone caused this in the work project I'm looking at now

Reply:
Interesting - haven't managed a repro on this, despite the 'handler' being super thin.   feel free to disregard and i'll keep trying to understand how/where it's failing.

in the meantime, `quiet` avoids the issue for us and keeps the console a bit clearer so that's ok

Reply:
Hey guys! Hopping in the chat because I'm having the same issue with v1.2.2:


<details>
<summary><b>👀 Console error</b></summary>

---------------------------------

```
index.js:1402 [MSW] 07:22:18 query AuthUser (200 OK)
index.js:1616 Uncaught (in promise) TypeError: Failed to execute 'text' on 'Response': body stream already read
    at serializeResponse (webpack-internal:///(app-client)/../../node_modules/.pnpm/msw@1.2.2_typescript@4.8.4/node_modules/msw/lib/index.js:1616:27)
    at Emitter.eval (webpack-internal:///(app-client)/../../node_modules/.pnpm/msw@1.2.2_typescript@4.8.4/node_modules/msw/lib/index.js:1660:25)
    at Emitter.onceListener (webpack-internal:///(app-client)/../../node_modules/.pnpm/strict-event-emitter@0.4.6/node_modules/strict-event-emitter/lib/Emitter.js:40:22)
    at eval (webpack-internal:///(app-client)/../../node_modules/.pnpm/strict-event-emitter@0.4.6/node_modules/strict-event-emitter/lib/Emitter.js:75:22)
    at Array.forEach (<anonymous>)
    at Emitter.emit (webpack-internal:///(app-client)/../../node_modules/.pnpm/strict-event-emitter@0.4.6/node_modules/strict-event-emitter/lib/Emitter.js:74:19)
    at source.emit (webpack-internal:///(app-client)/../../node_modules/.pnpm/msw@1.2.2_typescript@4.8.4/node_modules/msw/lib/index.js:2036:20)
    at eval (webpack-internal:///(app-client)/../../node_modules/.pnpm/msw@1.2.2_typescript@4.8.4/node_modules/msw/lib/index.js:1756:23)
    at ServiceWorkerContainer.eval (webpack-internal:///(app-client)/../../node_modules/.pnpm/msw@1.2.2_typescript@4.8.4/node_modules/msw/lib/index.js:2140:17)
serializeResponse @ index.js:1616
eval @ index.js:1660
onceListener @ Emitter.js:40
eval @ Emitter.js:75
emit @ Emitter.js:74
source.emit @ index.js:2036
eval @ index.js:1756
eval @ index.js:2140
await in eval (async)
onceListener @ Emitter.js:40
eval @ Emitter.js:75
emit @ Emitter.js:74
source.emit @ index.js:2036
eval @ index.js:1756
eval @ index.js:2140

```

---------------------------------

</details>

The following solutions worked for me:

1. use option `quiet: true`
2. downgrade to v1.2.1

With v1.2.2 and quiet set to false, the console error occurs on the second request to the same endpoint. In our case we are making two requests to an `AuthUser` query in GraphQL. The first request is logged to the console and the second request-log is followed by the error. That might be why the error says "body stream already read" — I hope that's helpful!

I'm on a work repo so I can't provide a reproduction env easily right now. That said, our implementation isn't special. I am seeing the console error while mocking responses (in local development) to our GraphQL API within a Next.js app.